outport/process/outportDataProvider.go

```diff
@@ -189,6 +195,65 @@ func (odp *outportDataProvider) PrepareOutportSaveBlockData(arg ArgPrepareOutpor
 	return outportBlock, nil
 }
 
+func (odp *outportDataProvider) prepareExecutionResultsData(args ArgPrepareOutportSaveBlockData) (map[string]*outportcore.ExecutionResultsData, error) {
+	if !args.Header.IsHeaderV3() {
+		return nil, nil
+	}
+
+	results := make(map[string]*outportcore.ExecutionResultsData)
+	for _, executionResult := range args.Header.GetExecutionResultsHandlers() {
+		headerHash := executionResult.GetHeaderHash()
+
+		body, err := common.GetCachedBody(odp.dataPool.ExecutedMiniBlocks(), odp.marshaller, executionResult)
+		if err != nil {
+			return nil, err
+		}
+		intraMbs, err := common.GetCachedIntraMbs(odp.dataPool.ExecutedMiniBlocks(), odp.marshaller, headerHash)
+		if err != nil {
+			return nil, err
+		}
+
+		cachedTxs, err := common.GetCachedIntermediateTxs(odp.dataPool.PostProcessTransactions(), headerHash)
+		if err != nil {
+			return nil, err
+		}
+		cachedLogs, err := common.GetCachedLogs(odp.dataPool.PostProcessTransactions(), headerHash)
+		if err != nil {
+			return nil, err
+		}
+		pool, err := odp.createPoolWithWrappedTxs(cachedTxs, cachedLogs)
+		if err != nil {
+			return nil, err
+		}
+
+		err = odp.transactionsFeeProcessor.PutFeeAndGasUsed(pool, executionResult.GetHeaderEpoch())
```

Comment:
Just a question:
This is called once in `PrepareOutportSaveBlockData` function and then for each execution result. Is that right?